internal/desc/desc.go

```diff
@@ -54,3 +55,50 @@ func SortFileDescriptorSet(fileDescriptorSet *descriptor.FileDescriptorSet, file
 	newFileDescriptorSet.File = append(newFileDescriptorSet.File, fileDescriptorProto)
 	return newFileDescriptorSet, nil
 }
+
+// SortFileDescriptorSet sorts a FileDescriptorSet in place by using its
+// relative import path and file name retrieved from GetName().
+//
+// This assumes that the give FileDescriptorSet is valid.
+func SortFileDescriptorSet(fileDescriptorSet *descriptor.FileDescriptorSet) {
+	sort.Slice(fileDescriptorSet.File, func(i, j int) bool {
+		return fileDescriptorSet.File[i].GetName() < fileDescriptorSet.File[j].GetName()
+	})
+}
+
+// MergeFileDescriptorSets deduplicates and sorts multiple file descriptor sets
+// into a single set.
+func MergeFileDescriptorSets(sets []*descriptor.FileDescriptorSet) *descriptor.FileDescriptorSet {
+	// Here we rely soley on GetName() to uniquely identify each
+	// FileDescriptorProto. GetName() returns the unique proto file name,
+	// including the import path from the current directory.
+	//
+	// eg: GetName() would return "bar/bar.proto" and "foo/foo.proto",
+	// respectively, for bar.proto and foo.proto in the following directory
+	// structure.
+	//
+	//  .
+	//  ├── bar
+	//  │   └── bar.proto
+	//  ├── foo
+	//  │   └── foo.proto
+	//  └── prototool.yaml
+	files := map[string]*descriptor.FileDescriptorProto{}
+	for _, fileDescriptorSet := range sets {
+		for _, fileDescriptorProto := range fileDescriptorSet.File {
+			name := fileDescriptorProto.GetName()
```

Comment:
you should probably error if there are overlapping names

Reply:
or maybe not, as I expect there will be overlaps given the current setup, let's talk offline